File: 01Recursion/Recursion.java
//Windows NT 10.0.10586 "10"
public class Recursion{

	/*public static void main(String[]args){
		Recursion x = new Recursion();
		System.out.println(x.name());
		int test = 10;
		System.out.println("Guess: " + x.sqrt(100));
		System.out.println("Actual: 10");
	}*/

	public String name(){
		return "Almawaldi,Sammy";
	}

	public double sqrt(double n){
		if(n < 0){
			throw new IllegalArgumentException();
		}
		if(n == 0){
			return 0;
		}
		return sqrtFormula(n, 1);
	}

	private double sqrtFormula(double n, double guess){
		if(percentError(Math.pow(guess, 2), n) < 0.000001){
			return guess;
		}
		else{
			//System.out.println(guess);
			return sqrtFormula(n, (n / guess + guess) / 2);
		}
	}

	private double percentError(double guess, double actual){
		return Math.abs((guess - actual) / guess) * 100;
	}
}

File: 01Recursion/Recursion.py
import math

def main():
	print(name())
	print(sqrt(1))
	print(sqrt(2))
	print(sqrt(4))
	print(sqrt(100))
	print(sqrt(0.0000001))
	print(sqrt(-1))
	print(sqrt(-100))
	print(sqrt(0))

def name():
	return "Almawaldi, Sammy"

def sqrt(n):
	if n < 0:
		return "IllegalArguementException"
	if n == 0:
		return 0
	return sqrtFormula(n, 1)

def sqrtFormula(n, guess):
	if percentError(math.pow(guess, 2), n) < 0.000001:
		return guess
	else:
		return sqrtFormula(n, (n / guess + guess) / 2)

def percentError(guess, actual):
	return math.fabs((guess - actual) / guess) * 100

main()